nuxeo-runtime/nuxeo-runtime-test/src/main/java/org/nuxeo/runtime/test/NXRuntimeTestCase.java

```diff
@@ -376,14 +373,19 @@ protected void initUrls() throws Exception {
      * This happens for instance if a previous test had errors in its <code>setUp()</code>, because
      * <code>tearDown()</code> has not been called.
      */
-    protected void wipeRuntime() throws Exception {
+    protected void wipeRuntime() {
         // Make sure there is no active runtime (this might happen if an
         // exception is raised during a previous setUp -> tearDown is not called
         // afterwards).
         runtime = null;
         frameworkStarted = false;
         if (Framework.getRuntime() != null) {
-            Framework.shutdown();
+            try {
+                Framework.shutdown();
+            } catch (InterruptedException cause) {
+                Thread.currentThread().interrupt();
+                throw new RuntimeServiceException("Interrupted during shutdown", cause);
```

Comment:
Maybe use `ExceptionUtils#checkInterrupt` ?

Reply:
We could if it allowed passing a message to the exception to rethrow.
Probably an improvement for `ExceptionUtils`.

Reply:
I'm advising to move away from `checkInterrupt` and instead use an explicit `Thread.currentThread().interrupt()` and rethrow so that code smell tools detect the proper pattern.